GW2EIEvtcParser/EIData/ProfHelpers/Mesmer/ChronomancerHelper.cs

```diff
@@ -84,5 +86,175 @@ internal static bool IsKnownMinionID(int id)
+                foreach (EffectEvent effect in wellsOfPrecognition)
+                {
+                    (int, int) lifespan = ProfHelper.ComputeEffectLifespan(log, effect, 5000);
+                    var connector = new PositionConnector(effect.Position);
+                    replay.Decorations.Add(new CircleDecoration(false, 0, 240, lifespan, color.WithAlpha(0.5f).ToString(), connector).UsingSkillMode(skill));
+                    replay.Decorations.Add(new IconDecoration(ParserIcons.EffectWellOfPrecognition, CombatReplaySkillDefaultSizeInPixel, CombatReplaySkillDefaultSizeInWorld, 0.5f, lifespan, connector).UsingSkillMode(skill));
+                }
+            }
+            // Well of Precognition - Pulses
+            if (log.CombatData.TryGetEffectEventsBySrcWithGUIDs(player.AgentItem, new string[] { EffectGUIDs.ChronomancerWellOfPrecognitionPulse, EffectGUIDs.ChronomancerWellOfPrecognitionExplosion }, out IReadOnlyList<EffectEvent> wellsOfPrecognitionPulses))
+            {
+                var skill = new SkillModeDescriptor(player, Spec.Chronomancer, WellOfPrecognition, SkillModeCategory.ImportantBuffs);
+                foreach (EffectEvent effect in wellsOfPrecognitionPulses)
+                {
+                    int effectTimeStart = (int)effect.Time;
+                    int effectTimeEnd = effectTimeStart + 1000;
+                    var connector = new PositionConnector(effect.Position);
+                    replay.Decorations.Add(new CircleDecoration(false, effectTimeEnd, 240, (effectTimeStart, effectTimeEnd), color.WithAlpha(0.5f).ToString(), connector).UsingSkillMode(skill));
+                }
+            }
+
+            // Well of Senility
+            if (log.CombatData.TryGetEffectEventsBySrcWithGUID(player.AgentItem, EffectGUIDs.ChronomancerWellOfSenility, out IReadOnlyList<EffectEvent> wellsOfSenility))
+            {
+                var skill = new SkillModeDescriptor(player, Spec.Chronomancer, WellOfRecall_Senility);
+                foreach (EffectEvent effect in wellsOfSenility)
+                {
+                    (int, int) lifespan = ProfHelper.ComputeEffectLifespan(log, effect, 5000);
+                    var connector = new PositionConnector(effect.Position);
+                    replay.Decorations.Add(new CircleDecoration(false, 0, 240, lifespan, color.WithAlpha(0.5f).ToString(), connector).UsingSkillMode(skill));
+                    replay.Decorations.Add(new IconDecoration(ParserIcons.EffectWellOfSenility, CombatReplaySkillDefaultSizeInPixel, CombatReplaySkillDefaultSizeInWorld, 0.5f, lifespan, connector).UsingSkillMode(skill));
+
+                    // Well pulses - Hard coded because the effects don't have a Src
+                    int pulseTimeDelay = 0;
+                    for (int i = 0; i < 4; i++)
+                    {
+                        int effectTimeStart = (int)effect.Time + pulseTimeDelay;
```

Comment:
Same situation as Well of Action

Reply:
you mean if (effectTimeStart > lifespan.Item2) { break; } ? I have already added it